shaman/controllers/nodes/__init__.py

```diff
@@ -42,6 +43,14 @@ def index(self):
             resp[node.host] = node.__json__()
         return resp
 
+    @secure(basic_auth)
+    @expose(method='GET', content_type="text/plain")
+    def next(self):
+        next_node = get_next_node()
+        if not next_node:
+            abort(500, "There are no healthy chacra nodes available.")
```

Comment:
Sure, I debated this one myself. In the end I decided that if shaman can not deliver a chacra node when asked, then that is an error state. ``404`` tells me that I've hit the wrong endpoint, not "this is the right endpoint, I just don't have anything to give you". I don't have really strong feelings about this though.